### PR TITLE
JS: Two join-order improvements

### PR DESCRIPTION
**Pull request by erik-krogh**
While evaluating my some of my recent PRs I looked into the benchmarks that performed badly.  

So I looked into it, and found a few issues... and those issues had nothing to do with the code in the PRs :shrug: 

[Evaluation shows a decent improvement](https://github.com/dsp-testing/erik-krogh-dca/tree/run/joins-default-security/reports).

### CODE REVIEW

File: javascript/ql/src/semmle/javascript/frameworks/AngularJS/ServiceDefinitions.qll [hunk 1]
```diff
@@ -297,34 +297,29 @@ abstract private class CustomSpecialServiceDefinition extends CustomServiceDefin
 /**
  * Holds if `mce` defines a service of type `moduleMethodName` with name `serviceName` using the `factoryFunction` as the factory function.
  */
+bindingset[moduleMethodName]
 private predicate isCustomServiceDefinitionOnModule(
   DataFlow::CallNode mce, string moduleMethodName, string serviceName,
-  DataFlow::SourceNode factoryFunction
+  DataFlow::SourceNode factoryArgument
```
Comment (asgerf) [inline javascript/ql/src/semmle/javascript/frameworks/AngularJS/ServiceDefinitions.qll:303]: Couldn't this be typed as `DataFlow::Node` to preserve the original behavior?

**Review by asgerf — COMMENTED** [review]
Looks great!

By the way I often see DIL predicates with the `pos#` prefix coming from the ServiceDefinitions and DependencyInjections files, which I'm told are a symptom of something bad happening in the optimizer. There is something seriously fishy going on in those libraries.

File: javascript/ql/src/semmle/javascript/frameworks/AngularJS/ServiceDefinitions.qll (hunk 1 above)
Comment (erik-krogh) [reply javascript/ql/src/semmle/javascript/frameworks/AngularJS/ServiceDefinitions.qll]: I'll do another evaluation now that you've merged the SourceNode PR.

**Comment by erik-krogh** [comment]
> I'll do another evaluation now that you've merged the SourceNode PR.

[Looks good](https://github.com/dsp-testing/erik-krogh-dca/tree/run/joins-default-security-2/reports).

**Review by asgerf — APPROVED** [review]

**Pull request merged**